### PR TITLE
View port handler nonnil

### PR DESCRIPTION
**Pull request by jjatie**

### CODE REVIEW

**Comment by liuxuan30** [comment]
@danielgindi do you want to take these similar PRs about swift improvements?

**Comment by jjatie** [comment]
@liuxuan30 Now that you've looked at the other related PRs, is it okay for this to be merged?

**Comment by liuxuan30** [comment]
I‘m checking every day but can't guarantee I can review daily; I would try. Reviewing slowly right now,  but I can get some merged for sure.

**Comment by liuxuan30** [comment]
I suggest in the future, similar improvements like remove optional can be squash into one PR, each commit could be one class or one topic to review, to reduce dependency and conflicts and makes it easier for you.

**Comment by jjatie** [comment]
Fixed

File: Source/Charts/Renderers/Renderer.swift [hunk 1]
```diff
@@ -12,21 +12,23 @@
 import Foundation
 import CoreGraphics
 
+//@objc(ChartRenderer)
```
Comment (liuxuan30) [inline Source/Charts/Renderers/Renderer.swift]: what is this, commented out code or documentation? If it's doc, maybe use doc style comment like /* or other

File: Source/Charts/Renderers/XAxisRenderer.swift [hunk 2]
```diff
@@ -339,10 +329,7 @@ open class XAxisRenderer: AxisRendererBase
     
     @objc open func drawGridLine(context: CGContext, x: CGFloat, y: CGFloat)
     {
-        guard
-            let viewPortHandler = self.viewPortHandler
-            else { return }
-        
+
```
Comment (liuxuan30) [inline Source/Charts/Renderers/XAxisRenderer.swift]: empty line? Not sure if this is github display issue, but if old guard is deleted, no space line needed. (ignore if there is no empty line)

**Comment by liuxuan30** [comment]
generally it's good, but I need to take a look at viewport handler myself if nil is fine.

File: Source/Charts/Charts/ChartViewBase.swift [hunk 3]
```diff
@@ -221,8 +221,7 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
         _animator = Animator()
         _animator.delegate = self
 
-        _viewPortHandler = ViewPortHandler()
-        _viewPortHandler.setChartDimens(width: bounds.size.width, height: bounds.size.height)
+        _viewPortHandler = ViewPortHandler(width: bounds.size.width, height: bounds.size.height)
```
[the remark this answers is not in the document]
Comment (jjatie) [reply Source/Charts/Charts/ChartViewBase.swift]: In some cases it is, that is why the method still exists. In this case it is not useful, as the only thing that is happening is what already happens in the initializer I changed it to.

File: Source/Charts/Renderers/Renderer.swift (hunk 1 above)
Comment (jjatie) [reply Source/Charts/Renderers/Renderer.swift]: That shouldn't be commented.

File: Source/Charts/Renderers/XAxisRenderer.swift (hunk 2 above)
Comment (jjatie) [reply Source/Charts/Renderers/XAxisRenderer.swift]: I'll fix

File: Source/Charts/Charts/ChartViewBase.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (jjatie) [reply Source/Charts/Charts/ChartViewBase.swift]: This is the `initialize` method, so it does need to be inited here.

**Comment by jjatie** [comment]
Fixed based on comments

File: Source/Charts/Renderers/XAxisRendererHorizontalBarChart.swift [hunk 4]
```diff
@@ -29,10 +29,7 @@ open class XAxisRendererHorizontalBarChart: XAxisRenderer
     
     open override func computeAxis(min: Double, max: Double, inverted: Bool)
     {
-        guard let
-            viewPortHandler = self.viewPortHandler
-            else { return }
-        
+
```
Comment (liuxuan30) [inline Source/Charts/Renderers/XAxisRendererHorizontalBarChart.swift]: empty line?

File: Source/Charts/Renderers/XAxisRendererHorizontalBarChart.swift [hunk 5]
```diff
@@ -200,10 +195,7 @@ open class XAxisRendererHorizontalBarChart: XAxisRenderer
     
     open override func drawGridLine(context: CGContext, x: CGFloat, y: CGFloat)
     {
-        guard
-            let viewPortHandler = self.viewPortHandler
-            else { return }
-        
+
```
Comment (liuxuan30) [inline Source/Charts/Renderers/XAxisRendererHorizontalBarChart.swift]: empty line?

File: Source/Charts/Renderers/YAxisRenderer.swift [hunk 6]
```diff
@@ -218,10 +212,7 @@ open class YAxisRenderer: AxisRendererBase
         context: CGContext,
         position: CGPoint)
     {
-        guard
-            let viewPortHandler = self.viewPortHandler
-            else { return }
-        
+
```
Comment (liuxuan30) [inline Source/Charts/Renderers/YAxisRenderer.swift]: empty line?

File: Source/Charts/Renderers/YAxisRendererHorizontalBarChart.swift [hunk 7]
```diff
@@ -194,10 +185,7 @@ open class YAxisRendererHorizontalBarChart: YAxisRenderer
         context: CGContext,
         position: CGPoint)
     {
-        guard
-            let viewPortHandler = self.viewPortHandler
-            else { return }
-        
+
```
Comment (liuxuan30) [inline Source/Charts/Renderers/YAxisRendererHorizontalBarChart.swift]: empty line?

File: Source/Charts/Utils/ViewPortHandler.swift [hunk 8]
```diff
@@ -54,11 +54,7 @@ open class ViewPortHandler: NSObject
     
     /// offset that allows the chart to be dragged over its bounds on the x-axis
     fileprivate var _transOffsetY = CGFloat(0.0)
-    
-    public override init()
-    {
-    }
-    
+
```
Comment (liuxuan30) [inline Source/Charts/Utils/ViewPortHandler.swift]: empty line?

File: Source/Charts/Renderers/XAxisRenderer.swift [hunk 9]
```diff
@@ -19,17 +19,14 @@ import CoreGraphics
...
         super.init(viewPortHandler: viewPortHandler, transformer: transformer, axis: xAxis)
     }
     
     open override func computeAxis(min: Double, max: Double, inverted: Bool)
     {
-        guard let
-            viewPortHandler = self.viewPortHandler
-            else { return }
-        
+
```
Comment (liuxuan30) [inline Source/Charts/Renderers/XAxisRenderer.swift]: empty line?

File: Source/Charts/Utils/ViewPortHandler.swift (hunk 8 above)
Comment (jjatie) [reply Source/Charts/Utils/ViewPortHandler.swift]: This one should be here. The rest are now fixed

**Comment by liuxuan30** [comment]
Finished reviewing. `ChartViewBase` guaranteed `viewPortHandler` will be initialized anyway, so renderer's subclass will always have `viewPortHandler`. Thus removing optional `?` and `!` seems reasonable.

**Pull request merged**